analytics-api/app/controllers/JobController.scala

```diff
@@ -30,48 +29,72 @@ class JobController @Inject() (
     val body: String = Json.stringify(request.body.asJson.get)
     val channelId = request.headers.get("X-Channel-ID").getOrElse("")
     val consumerId = request.headers.get("X-Consumer-ID").getOrElse("")
-    val checkFlag = if (config.getBoolean("dataexhaust.authorization_check")) authorizeDataExhaustRequest(consumerId, channelId) else true
-    if (checkFlag) {
-      val res = ask(jobAPIActor, DataRequest(body, channelId, config)).mapTo[Response]
-      res.map { x =>
-        result(x.responseCode, JSONUtils.serialize(x))
-      }
-    } else {
-      val msg = s"Given X-Consumer-ID='$consumerId' and X-Channel-ID='$channelId' are not authorized"
-      APILogger.log(s"Authorization FAILED for X-Consumer-ID='$consumerId' and X-Channel-ID='$channelId'")
-      unauthorized(msg)
+    if (channelId.nonEmpty) {
+        val checkFlag = if (config.getBoolean("dataexhaust.authorization_check")) authorizeDataExhaustRequest(consumerId, channelId) else true
+        if (checkFlag) {
+            val res = ask(jobAPIActor, DataRequest(body, channelId, config)).mapTo[Response]
+            res.map { x =>
+                result(x.responseCode, JSONUtils.serialize(x))
+            }
+        } else {
+            val msg = s"Given X-Consumer-ID='$consumerId' and X-Channel-ID='$channelId' are not authorized"
+            APILogger.log(s"Authorization FAILED for X-Consumer-ID='$consumerId' and X-Channel-ID='$channelId'")
+            unauthorized(msg)
+        }
+    }
+    else {
+        val msg = "X-Channel-ID is missing in request header"
+        APILogger.log("X-Channel-ID is missing in request header")
+        invalid(msg, APIIds.DATA_REQUEST)
```

Comment:
Code duplicated in multiple methods. Move this to a generic method.

Reply:
Refactored to have generic method